scripts/hedera-response-codes-protobuf-parser.js

```diff
@@ -0,0 +1,53 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2025 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const fs = require('fs');
+const axios = require('axios');
+const protobufjs = require('protobufjs');
+
+async function main() {
+  const version = (await axios.get('https://raw.githubusercontent.com/hashgraph/hedera-services/refs/heads/main/version.txt')).data.replace('\n', '');
+  const res = (await axios.get('https://raw.githubusercontent.com/hashgraph/hedera-services/refs/heads/main/hapi/hedera-protobufs/services/response_code.proto')).data;
+  const parsedProto = protobufjs.parse(res, {
+    keepCase: true,
+    alternateCommentMode: true,
+    preferTrailingComment: true
+  });
+  const responseCodes = parsedProto.root.nested.proto.nested.ResponseCodeEnum;
+
+  let contract =
+      `// SPDX-License-Identifier: Apache-2.0\n` +
+      `pragma solidity >=0.4.9 <0.9.0;\n` +
+      `\n// this contract is auto-generated by a manual triggered script in utils/hedera-response-codes-protobuf-parser.js` +
+      `\n// the generated contract is using hedera response codes from services version ${version}` +
+      `\n// https://github.com/hashgraph/hedera-services/blob/main/hapi/hedera-protobufs/services/response_code.proto\n\n` +
+      `library HederaResponseCodes {\n`;
+  for (const [name, code] of Object.entries(responseCodes.values)) {
+    contract += `    // ${responseCodes.comments[name]?.replaceAll('\n', ' ')}\n`;
```

Comment:
should we use NatSpec, like `///` or `/** */` for the comments? So people including this in their projects can see the docs in their IDEs?

Reply:
Fixed, using `///` for single-line comments.